projects/plugins/jetpack/tests/php/extensions/blocks/test-business-hours.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Business Hours Block tests
+ *
+ * @package automattic/jetpack
+ */
+
+/**
+ * Include the file containing the block's registration and render functions.
+ */
+require_once JETPACK__PLUGIN_DIR . 'extensions/blocks/business-hours/business-hours.php';
+
+const BLOCK_NAME = 'jetpack/business-hours';
+
+/**
+ * Business Hours Block tests.
+ *
+ * These tests primarily check that server rendered markup is not changing unexpectedly
+ * when serialized fixtures are updated via the block's JS-based save behaviour.
+ *
+ * The goal is to catch when changes to serialized markup affects server rendering of the block.
+ */
+class Business_Hours_Block_Test extends \WP_UnitTestCase {
+	/**
+	 * A variable to track whether or not the block was already registered before the test was run.
+	 *
+	 * @access private
+	 *
+	 * @var boolean
+	 */
+	private $was_registered = false;
+
+	/**
+	 * Setup and ensure the block is registered before running the tests.
+	 *
+	 * @before
+	 */
+	public function set_up() {
+		$this->was_registered = \Automattic\Jetpack\Blocks::is_registered( BLOCK_NAME );
+		\Automattic\Jetpack\Extensions\Business_Hours\register_block();
+	}
+
+	/**
+	 * Teardown and unregister the block if it wasn't registered before running these tests.
+	 *
+	 * @after
+	 */
+	public function tear_down() {
+		if ( ! $this->was_registered ) {
+			unregister_block_type( 'jetpack/business-hours' );
+		}
+	}
+
+	/**
+	 * Test that the block is registered, which means that it can be registered.
+	 */
+	public function test_block_can_be_registered() {
+		$is_registered = \Automattic\Jetpack\Blocks::is_registered( BLOCK_NAME );
+		$this->assertTrue( $is_registered );
+	}
+
+	/**
+	 * This test iterates over the block's serialized fixtures, and tests that the generated
+	 * markup matches a fixture for the server rendered markup for the block.
+	 *
+	 * If no server rendered fixture can be found, then one is created.
+	 */
+	public function test_server_side_rendering_based_on_serialized_fixtures() {
+		// phpcs:disable WordPress.WP.AlternativeFunctions
+		$fixtures_path = 'extensions/blocks/business-hours/test/fixtures/';
+		$file_pattern  = '*.serialized.html';
+		$files         = glob( JETPACK__PLUGIN_DIR . $fixtures_path . $file_pattern );
```

Comment:
Not for this PR, but if we get the 🟢 for these tests this and some of the surrounding code blocks would be handy as a shared util or even custom assertion method.

Reply:
Thanks, I was thinking of that, too. I wasn't quite sure how much each block might vary in their implementation, so thought refactoring to utils could be a good thing to do if and when this one's approved and we work on the second (or third) test.